### PR TITLE
Add Avalon FAQ

### PR DESCRIPTION
**Pull request by danintel**
Frequently-asked questions with answers.

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by danintel** [comment]
Here is a link to the rendered markup, which is easier to read:
https://github.com/danintel/avalon/blob/faq/docs/FAQ.rst

File: docs/FAQ.rst [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+Is there a maximum size of input data when using Intel SGX?
+-----------------------------------------------------------
+Avalon doesn’t expect application-specific code to use SGX sealed data.
+Avalon uses sealed data internally for storing private enclave signing and
+encryption keys.
+As result application specific data size is not dependent on the sealed data
+storage.
+It is indirectly limited by the maximum enclave size
+(enclave includes both data and code).
+The maximum enclave size is limited to 128Mbytes.
```
Comment (manojgop) [inline docs/FAQ.rst]: EPC size depends on the platform.  Fore example, Intel Xeon E Mehlow-R has 256MB EPC, Xeon E3 is 128MB.

File: docs/FAQ.rst [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+For details see:
+https://lists.hyperledger.org/g/avalon/topic/68762393
+or
+https://wiki.hyperledger.org/display/avalon/Hyperledger+Avalon
+
+Where are videos about Hyperledger Avalon?
+------------------------------------------
+- `Hyperledger Avalon Heart Disease Demo (2019, 10:25) <https://youtu.be/6L_UOhi7Rxs>`_
+- `iExec, Microsoft and Intel present the TCF Trusted Compute Framework at
+  Devcon (EEA token & other uses) (2019, 1:38:18) <https://youtu.be/lveTxAQ6rmQ>`_
```
[the remark this answers is not in the document]
Comment (danintel) [reply docs/FAQ.rst]: I did not know about these. I added the links.
I also created a "Hyperledger Avalon" playlist on YouTube, to aid others finding the YouTube-hosted videos, with 3 videos.

File: docs/FAQ.rst [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  https://github.com/hyperledger/avalon/blob/master/BUILD.md
+- Intel SGX is not required to build or use Avalon.
+
+Where are the Avalon libraries?
+-------------------------------
+Avalon libraries are provided in source form at
+https://github.com/hyperledger/avalon/
+and must be compiled.
+The enclave libraries are in subdirectory ``tc/sgx/`` .
+Client libraries are under ``examples/common``
```
[the remark this answers is not in the document]
Comment (danintel) [reply docs/FAQ.rst]: Agreed. That needs to be done after it's integrated.

File: docs/FAQ.rst [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+Docker and Containers
+=====================
+
+What is the purpose of using Docker, and what does it have to do with Intel SGX? I mean, are enclaves running in containers?
+----------------------------------------------------------------------------------------------------------------------------
+Docker is used only as a convenience. It has nothing to do with the Avalon
+security model or enclaves. Docker makes it easier to build and setup Avalon,
+but you can also build without Docker
+(although it requires several more steps).
```
[the remark this answers is not in the document]
Comment (danintel) [reply docs/FAQ.rst]: I clarified by changing "enclaves" to "Intel SGX enclaves"

File: docs/FAQ.rst (hunk 1 above)
Comment (danintel) [reply docs/FAQ.rst]: I do not want to mention possible future products here. I'll add Menlow.

Comment (danintel) [reply docs/FAQ.rst]: I dropped the changes for this PR.  See PR #242 for changes.

**Pull request merged**